src/dxmt/dxmt_command_queue.hpp

```diff
@@ -356,11 +356,11 @@ class CommandQueue {
   };
 
   uint64_t CurrentSeqId() {
-    return ready_for_encode.load(std::memory_order_relaxed);
+    return ready_for_encode.load(std::memory_order_acquire);
```

Comment:
Then we can talk about the atomics

- `ready_for_encode`: it's exclusively written by rendering thread. it can be read by rendering thread but doesn't require any atomicity. the encoding thread waits on it to cap its internal seq_id (the encoding thread doesn't read its value directly because the values of two loads may be not the same or successive, but with a "gap")

- `ready_for_commit`: actually the name is kinda misleading, should be `ready_for_finish`. and it operates similarly to `ready_for_encode`, just between the encoding thread and finishing thread.

- `cpu_coherent`: it tells that all chunk id before/equal to is coherent.

all three above are directly related to the concept of chunk id.

- `chunk_ongoing` :  it's a value in range of [0, kCommandChunkCount], meaning how many chunks are occupied by the encoding and finishing thread. it's incremented by the rendering thread and decremented by the finishing thread. if it's equal to `kCommandChunkCount`, it will cause the rendering thread to wait.
> I guess `chunk_ongoing.wait(kCommandChunkCount - 1, std::memory_order_relaxed);` is not correct and should remove the `- 1`

`chunk_ongoing = ready_for_encode - cpu_coherent` so in theory we only need three atomics....